packages/firestore/test/integration/api/batch_writes.test.ts

```diff
@@ -317,36 +319,39 @@ apiDescribe('Database batch writes', persistence => {
     });
   });
 
-  it('can update nested fields', () => {
-    const initialData = {
-      desc: 'Description',
-      owner: { name: 'Jonny' },
-      'is.admin': false
-    };
-    const finalData = {
-      desc: 'Description',
-      owner: { name: 'Sebastian' },
-      'is.admin': true
-    };
+  (!persistence && USE_EMULATOR ? it.skip : it)(
```

Comment:
`(!persistence && USE_EMULATOR ? it.skip : it)` is a bit cumbersome, but it seems to be the way we exclude tests currently like [here](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/test/integration/api/database.test.ts#L119-L141) or [here](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/test/unit/local/simple_db.test.ts#L353-L371). I wonder if we want to have some customized runner, like what we have for firestore unit test [here](https://github.com/firebase/firebase-js-sdk/blob/master/packages/firestore/test/unit/specs/describe_spec.ts#L81-L102).

Reply:
Why does `persistence` come into play here?  Also:
1. Please add comments (ideally in the form of something like "TODO(b/...): Disabled because emulator currently ..."
2. I don't trust my operator precedence assumptions, so if possible I'd throw some extra parents around `(!persistence && USE_EMULATOR)`.

Reply:
It is because that these tests failed only when persistence is false. I have categorized all the failure symptoms/messages in this [doc](https://docs.google.com/document/d/1NF9DGH1bL9OIxQk207aMkA1MOC_zJWOO7uHcizao41U). I can add some comments in code, but I'm not sure whether these failures are due to the sdk test itself or the emulator.

There is also a sample travis [run](https://travis-ci.org/firebase/firebase-js-sdk/builds/467276768) that shows these failures (break down to each test file).

Reply:
I have tried to debug some failures, but couldn't tell whether it is because of the emulator itself, or we are calling the emulator in a wrong way, or there's more setup for the emulator that needs to be done before any test. I'll add more information I found in the doc.

So for this PR, do you think if it makes sense if I
* remove code for 1) excluding any tests, 2) including firestore emulator test on Travis;
* keep code for 1) starting/stoping emulators (a bit of refactoring), 2) yarn command to run database/firestore test with their emulators.

I want to have some partial result completed before the quarter ends, and it's probably easier for you to help debug failures this way.

Reply:
Sure, that sounds great.